src/web/handlers/FlingGestureHandler.ts

```diff
@@ -1,30 +1,73 @@
 import { State } from '../../State';
 import { Direction } from '../constants';
 import { AdaptedEvent, Config } from '../interfaces';
+import PointerTracker from '../tools/PointerTracker';
 
 import GestureHandler from './GestureHandler';
 
 const DEFAULT_MAX_DURATION_MS = 800;
-const DEFAULT_MIN_ACCEPTABLE_DELTA = 32;
+const DEFAULT_MIN_VELOCITY = 700;
+const DEFAULT_MIN_DIRECTION_ALIGNMENT = 0.75;
 const DEFAULT_DIRECTION = Direction.RIGHT;
 const DEFAULT_NUMBER_OF_TOUCHES_REQUIRED = 1;
 
+class Vector {
+  x: number = 0;
+  y: number = 0;
+  uX: number = 0;
+  uY: number = 0;
+
+  fromDirection(direction: number) {
+    [this.x, this.y] = [this.uX, this.uY] = new Map([
+      [Direction.LEFT, [-1, 0]],
+      [Direction.RIGHT, [1, 0]],
+      [Direction.UP, [0, -1]],
+      [Direction.DOWN, [0, 1]],
+    ]).get(direction) ?? [0, 0];
+
+    return this;
+  }
+
+  fromVelocity(tracker: PointerTracker, pointerId: number) {
+    this.x = tracker.getVelocityX(pointerId);
+    this.y = tracker.getVelocityY(pointerId);
+
+    const magnitude = Math.hypot(this.x, this.y);
+    if (magnitude < 0.001) {
+      this.uX = this.uY = 0;
+    }
+
+    this.uX = this.x / magnitude;
+    this.uY = this.y / magnitude;
+
+    return this;
+  }
+
+  computeSimilarity(vector: any) {
```

Comment:
```suggestion
  computeSimilarity(vector: Vector) {
```

I know there are places in our codebase where we still use `any`, but here we can simply use `Vector`.